postgrescontroller/pkg/stub/handler.go

```diff
@@ -0,0 +1,587 @@
+			break
+		} else {
+			fmt.Println("Waiting for Pod to get ready.")
+			time.Sleep(time.Second * 4)
+		}
+	}
+
+	time.Sleep(time.Second * 2)
+
+	if len(userAndDBCommands) > 0 {
+		fmt.Printf("About to create temp db file for user and db commands")
+		//file := createTempDBFile(userAndDBCommands)
+		fmt.Println("Now setting up the database")
+		//setupDatabase_prev(serviceIP, servicePort, file)
+		var dummyList []string
+		setupDatabase(serviceIP, servicePort, userAndDBCommands, dummyList)
+	}
+
+	if len(setupCommands) > 0 {
+		fmt.Printf("About to create temp db file for setup commands")
+		//file := createTempDBFile(setupCommands)
+		fmt.Println("Now setting up the database")
+		//setupDatabase(serviceIP, servicePort, file)
+		setupDatabase(serviceIP, servicePort, setupCommands, databases)
+	}
+
+	verifyCmd := strings.Fields("psql -h " + serviceIP + " -p " + nodePort + " -U <user> " + " -d <db-name>")
+	var verifyCmdString = strings.Join(verifyCmd, " ")
+	fmt.Printf("VerifyCmd: %v\n", verifyCmd)
+	return serviceIP, servicePort, allCommands, databases, users, verifyCmdString, err
+
+}
+
+func setupDatabase(serviceIP string, servicePort string, setupCommands []string, databases []string) {
+
+	fmt.Println("Setting up database")
+	fmt.Println("Commands:")
+	fmt.Printf("%v", setupCommands)
+
+	var host = serviceIP
+	port := -1
+	port, _ = strconv.Atoi(servicePort)
+	var user = "postgres"
+	var password = PGPASSWORD
+
+	var psqlInfo string
+	if len(databases) > 0 {
+		dbname := databases[0]
+		fmt.Println("%s\n", dbname)
+		psqlInfo = fmt.Sprintf("host=%s port=%d user=%s "+
+			"password=%s dbname=%s sslmode=disable",
+			host, port, user, password, dbname)
+	} else {
+		psqlInfo = fmt.Sprintf("host=%s port=%d user=%s "+
+			"password=%s sslmode=disable",
+			host, port, user, password)
+	}
+
+	db, err := sql.Open("postgres", psqlInfo)
+	if err != nil {
+		panic(err)
+	}
+	defer db.Close()
+
+	err = db.Ping()
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Successfully connected!")
+
+	for _, command := range setupCommands {
+		_, err = db.Exec(command)
+		if err != nil {
+			panic(err)
+		}
+	}
+	fmt.Println("Done setting up the database")
+}
+
+func createTempDBFile(setupCommands []string) (*os.File){
+	file, err := ioutil.TempFile("/tmp", "create-db1")
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("Database setup file:%s\n", file.Name())
+
+	for _, command := range setupCommands {
+		//fmt.Printf("Command: %v\n", command)
+		// TODO: Interpolation of variables
+		file.WriteString(command)
+		file.WriteString("\n")
+	}
+	file.Sync()
+	file.Close()
+	return file
+}
+
+// podList returns a v1.PodList object
+func getPods() *apiv1.PodList {
+	return &apiv1.PodList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+	}
+}
+
+func int32Ptr(i int32) *int32 { return &i }
+
+// newbusyBoxPod demonstrates how to create a busybox pod
+func newbusyBoxPod(cr *v1.Postgres) *corev1.Pod {
```

Comment:
I don't think this function is used. Please verify and delete if not being used.